lib/ansible/modules/cloud/azure/azure.py

```diff
@@ -305,7 +306,7 @@ def _handle_timeout(signum, frame):
                     azure.delete_disk(disk.name, True)
                     disk_names.remove(disk_name)
     except AzureException as e:
-        module.fail_json(msg="failed to get or delete disk %s, error was: %s" % (disk_name, str(e)))
+        raise AzureException("failed to get or delete disk %s, error was: %s" % (disk_name, str(e)))
```

Comment:
this is  not a fix, module should always exist gracefully (not in exception) .. the str() should be changed to_native though